Thanks for the fixes @atillack. 

I've tested commit `396556a` with simple commands, without _yet_ using long paths, just as a preliminary test. The .dlg file is produced, but core is dumped:

* `make DEVICE=GPU test` and  `make DEVICE=OCLGPU test` produce similar outputs:

```bash
[...]
./bin/autodock_gpu_64wi \
-ffile ./input/3ce3/derived/3ce3_protein.maps.fld \
-lfile ./input/3ce3/derived/3ce3_ligand.pdbqt \
-nrun 100 \
-ngen 27000 \
-psize 150 \
-resnam test \
-gfpop 0 \
-lsmet sw
AutoDock-GPU version: v1.3-2-g396556a773c2a1c153232228f35baaeedddb2dbb

Kernel source used for development:      ./device/calcenergy.cl                  
Kernel string used for building:         ./host/inc/stringify.h                  
Kernel compilation flags:                 -I ./device -I ./common -DN64WI   -cl-mad-enable
OpenCL device:                           GeForce RTX 2070 SUPER
(Thread 0 is setting up Job 0)

Running Job #0:
Local-search chosen method is: Solis-Wets (sw)

Executing docking runs:
        20%        40%       60%       80%       100%
---------+---------+---------+---------+---------+
**************************************************

Job #0 took 9.211 sec after waiting 0.356 sec for setup

(Thread 0 is processing Job 0)

Run time of entire job set (1 file): 9.896 sec
corrupted size vs. prev_size in fastbins
make: *** [Makefile.OpenCL:266: test] Aborted (core dumped)
```

* `make DEVICE=CPU NRUN=1 test`

```
[...]
./bin/autodock_cpu_16wi \
-ffile ./input/3ce3/derived/3ce3_protein.maps.fld \
-lfile ./input/3ce3/derived/3ce3_ligand.pdbqt \
-nrun 1 \
-ngen 27000 \
-psize 150 \
-resnam test \
-gfpop 0 \
-lsmet sw
AutoDock-GPU version: v1.3-2-g396556a773c2a1c153232228f35baaeedddb2dbb

Kernel source used for development:      ./device/calcenergy.cl                  
Kernel string used for building:         ./host/inc/stringify.h                  
Kernel compilation flags:                 -I ./device -I ./common -DN16WI   -cl-mad-enable
OpenCL device:                                  Intel(R) Core(TM) i5-2450M CPU @ 2.50GHz
(Thread 0 is setting up Job 0)

Running Job #0:
Local-search chosen method is: Solis-Wets (sw)

Executing docking runs:
        20%        40%       60%       80%       100%
---------+---------+---------+---------+---------+
**************************************************

Job #0 took 32.167 sec after waiting 2.841 sec for setup

(Thread 0 is processing Job 0)
*** Error in `./bin/autodock_cpu_16wi': double free or corruption (fasttop): 0x0000000002e70ab0 ***
======= Backtrace: =========
/lib/x86_64-linux-gnu/libc.so.6(+0x777f5)[0x7fa7151607f5]
/lib/x86_64-linux-gnu/libc.so.6(+0x8038a)[0x7fa71516938a]
/lib/x86_64-linux-gnu/libc.so.6(cfree+0x4c)[0x7fa71516d58c]
./bin/autodock_cpu_16wi[0x403e26]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf0)[0x7fa715109840]
./bin/autodock_cpu_16wi[0x404339]
======= Memory map: ========
00400000-00464000 r-xp 00000000 08:08 7080461                            /home/lvs/Desktop/AutoDock-GPU/bin/autodock_cpu_16wi
[...]
ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]
Makefile.OpenCL:266: recipe for target 'test' failed
make: *** [test] Aborted (core dumped)
```

The `develop` branch works without problems on both platforms. Can you please check this ?

--- a/host/src/getparameters.cpp
+++ b/host/src/getparameters.cpp
@@ -285,7 +285,7 @@ void get_commandpars(const int* argc,
 	mypars->handle_symmetry		= true;
 	mypars->gen_finalpop		= false;
 	mypars->gen_best		= false;
-	mypars->resname = strdup("docking");
+	mypars->resname			= strdup("docking");
 	mypars->qasp			= 0.01097f;
 	mypars->rmsd_tolerance 		= 2.0;			//2 Angstroem
 	mypars->xrayligandfile = strdup(mypars->ligandfile);	// By default xray-ligand file is the same as the randomized input ligand
@@ -678,7 +678,7 @@ void get_commandpars(const int* argc,
 		if (strcmp("-asfreq", argv [i]) == 0)
 		{
 			arg_recognized = 1;
-			sscanf(argv [i+1], "%u", &tempint);
+			sscanf(argv [i+1], "%ld", &tempint);
 			if ((tempint >= 1) && (tempint <= 100))
 				mypars->as_frequency = (unsigned int) tempint;
 			else

--- a/host/src/main.cpp
+++ b/host/src/main.cpp
@@ -109,8 +109,8 @@ int main(int argc, char* argv[])
 	{
 		if (strcmp("-devnum", argv [i]) == 0)
 		{
-			int tempint;
-			sscanf(argv [i+1], "%lu", &tempint);
+			unsigned int tempint;
+			sscanf(argv [i+1], "%u", &tempint);
 			if ((tempint >= 1) && (tempint <= 65536))
 				cData.devnum = (unsigned long) tempint-1;
 			else

--- a/host/src/processgrid.cpp
+++ b/host/src/processgrid.cpp
@@ -41,7 +41,7 @@ int get_gridinfo(const char* fldfilename, Gridinfo* mygrid)
 
 	char* ts1 = strdup(fldfilename);
 	#ifndef _WIN32
-	mygrid->grid_file_path = dirname(ts1);
+	mygrid->grid_file_path = strdup(dirname(ts1));
 	#else
 	char drive_tmp[_MAX_DRIVE];
 	char path_tmp[_MAX_DIR];

--- a/host/src/setup.cpp
+++ b/host/src/setup.cpp
@@ -158,10 +158,14 @@ int setup(std::vector<Map>& all_maps,
 		mypars.resname = (char*)malloc((filelist.max_len+1)*sizeof(char));
 		strcpy(mypars.resname, filelist.resnames[i_file].c_str());
 	} else if (filelist.used) { // otherwise add the index to existing name distinguish the files if multiple
+		char* tmp = strdup(mypars.resname);
+		char* nrtmp = strdup(std::to_string(i_file).c_str());
 		free(mypars.resname);
-		mypars.resname = (char*)malloc((filelist.max_len+11)*sizeof(char));
-		std::string if_str = std::to_string(i_file);
-		strcat(mypars.resname, if_str.c_str());
+		mypars.resname = (char*)malloc((strlen(tmp)+strlen(nrtmp)+1)*sizeof(char));
+		strcpy(mypars.resname, tmp);
+		strcat(mypars.resname, nrtmp);
+		free(tmp);
+		free(nrtmp);
 	}
 
 	Gridinfo mydummygrid;

--- a/host/src/setup.cpp
+++ b/host/src/setup.cpp
@@ -46,6 +46,9 @@ int preload_gridsize(FileList& filelist)
 		if(curr_size>gridsize)
 			gridsize=curr_size;
 	}
+	if(mygrid.grid_file_path) free(mygrid.grid_file_path);
+	if(mygrid.receptor_name) free(mygrid.receptor_name);
+	if(mygrid.map_base_name) free(mygrid.map_base_name);
 	return gridsize;
 }
 
@@ -151,11 +154,13 @@ int setup(std::vector<Map>& all_maps,
 	get_commandpars(&argc, argv, &(mygrid.spacing), &mypars);
 
 	if (filelist.resnames.size()>0){ // Overwrite resname with specified filename if specified in file list
-		mypars.resname = (char*)realloc(mypars.resname,(filelist.max_len+11)*sizeof(char));
+		free(mypars.resname);
+		mypars.resname = (char*)malloc((filelist.max_len+1)*sizeof(char));
 		strcpy(mypars.resname, filelist.resnames[i_file].c_str());
 	} else if (filelist.used) { // otherwise add the index to existing name distinguish the files if multiple
+		free(mypars.resname);
+		mypars.resname = (char*)malloc((filelist.max_len+11)*sizeof(char));
 		std::string if_str = std::to_string(i_file);
-		mypars.resname = (char*)realloc(mypars.resname,(filelist.max_len+11)*sizeof(char));
 		strcat(mypars.resname, if_str.c_str());
 	}
 
@@ -291,6 +296,7 @@ int load_all_maps (const char* fldfilename, const Gridinfo* mygrid, std::vector<
 
 		fclose(fp);
 	}
+	free(tempstr);
 	return 0;
 }
 

